src/app/games/pentago/PentagoMove.ts

```diff
@@ -12,19 +12,20 @@ export class PentagoMove extends MoveCoord {
                 coord: Coord.encoder.encode(move.coord),
             };
             if (move.blockTurned.isPresent()) {
-                encoded['blockTurned'] = move.blockTurned.get();
-                encoded['turnedClockwise'] = move.turnedClockwise;
+                encoded.blockTurned = move.blockTurned.get();
+                encoded.turnedClockwise = move.turnedClockwise;
             }
 
             return encoded;
         }
         public decodeMove(encoded: JSONValueWithoutArray): PentagoMove {
             const casted: JSONObject = encoded as JSONObject;
 
-            const coord: Coord = Coord.encoder.decode(casted['coord']);
-            const nullableBlockTurned: MGPOptional<number> = MGPOptional.ofNullable(casted['blockTurned'] as number | null);
+            const coord: Coord = Coord.encoder.decode(casted.coord);
+            const nullableBlockTurned: MGPOptional<number> =
+                MGPOptional.ofNullable(casted.blockTurned as number | null);
```

Comment:
blockTurned does not exist in JSONObject oO ?

Reply:
tout existe dans un jsonObject voyons ! (en vrai, oui : il peut être null cependant)

Reply:
Mais l'idée c'est pas justement de préfer les choses typées que de faire du vieux js comme jsonObject.toutEtNImporteQuoi ?

Reply:
Tout à fait, mais par moment tu peux pas avoir de garantie sur ton type, typiquement ici t'as un élément qui vient de firebase en JSON et le langage peut pas te garantir qu'il contient les bons champs. Note que le | null vient bien dire « c'est potentiellement null », et le ofNullable enlève le null car on sait qu'il ne peut pas l'être (et s'il l'est, ça throw à cet endroit).

Reply:
Oui mais du coup, on utilise a['b'] dans ces cas là justement, non ? a.b donne l'impression mensongère que c'est typé alors que c'est le linter qui fait sa souplette !

Reply:
a['b'] et a.b sont typés exactement de la même manière ici hein ! L'un est juste un raccourci pour l'autre. Mais donc ici tu préférerais a['b']?

Reply:
Pour moi a['b'] c'est quand a n'as pas vraiment une signature propre et claire, mais là si ça a été casté j'imagine qu'autant faire a.b ? Si on caste du non castable ceci dit, ça fait quoi ?

Reply:
caster du non castable, genre `'foo' as number`, ça te dit « éh tu peux pas faire ça ! Si t'es vraiment sûr fait : `'foo' as unknown as number`